crates/compiler/fmt/src/collection.rs

```diff
@@ -176,3 +181,134 @@ fn has_comments(spaces: &[CommentOrNewline<'_>]) -> bool {
     }
     false
 }
+
+pub(crate) enum RecordPrefix<'a, Field: Formattable + std::fmt::Debug> {
+    Update(&'a Loc<Field>),
+    Mapper(&'a Loc<Field>),
+}
+
+pub(crate) fn fmt_record_like<'a, 'b: 'a, Field, ToSpacesAround>(
```

Comment:
Eventually, it might be nice to use this for other types of collections like lists. I don't think we need to make any changes right now, but food for thought for the future.